docs/usage/integration.md

```diff
@@ -0,0 +1,314 @@
+contrasted with a more conventional database.
+
+Strict consistency guarantees (at the database level) simplify application logic and error handling
+farther up the stack.
+
+
+
+## Consistency
+
+TigerBeetle provides strict serializability
+([serializability + linearizability](http://www.bailis.org/blog/linearizability-versus-serializability/))
+to each [client session](#client-sessions).
+
+But consistency models can seem arcane.
+What specific guarantees does TigerBeetle provide to applications?
+
+- [**Client sessions**](#client-sessions)
+  - A client session may have at most one in-flight request.
+  - A client session reads its own writes.
+  - A client session observes writes in the order that they occur on the cluster.
+  - A client session observes [`debits_posted`](../reference/accounts.md#debits_posted) and
+    [`credits_posted`](../reference/accounts.md#credits_posted) as monotonically increasing.
+    That is, a client session will never see `credits_posted` or `debits_posted` decrease.
+  - A client session never observes uncommitted updates.
+  - A client session never observes a broken invariant (e.g.
+    [`flags.credits_must_not_exceed_debits`](../reference/accounts.md#flags.credits_must_not_exceed_debits)).
+  - Multiple client sessions may receive replies [out of order](#reply-order) relative to one another.
+  - A client session can consider a request executed when it receives a reply for the request.
+- [**Requests**](#requests)
+  - A request executes within the cluster at most once.
+  - Requests do not [time out](#retries) — a timeout typically implies failure, which cannot be
+    conclusively determined in the context of network faults.
+  - Requests retried by their original client session receive identical replies.
+  - Requests retried by a different client (same request body, different session) may receive
+    [different replies](#consistency-with-foreign-databases).
+  - Operations within a request are executed in sequential order.
+  - Operations within a request do not interleave with operations from other requests.
+    (TODO: Can timeouts interleave batches, or should we shift the batch so that timeouts land
+    entirely before/after?)
+- **Operations**
+  - Once committed, an operation will always be committed — the cluster's state never backtracks.
+  - [Accounts](../reference/accounts.md) are immutable — once created, they are never modified
+    (excluding balance fields, which are modified by transfers).
+  - [Transfers](../reference/transfers.md) are immutable — once created, they are never modified.
+  - Transfer [timeouts](../reference/transfers.md#timeout) are deterministic, driven by the
+    cluster's timestamp.
+  - Object timestamps are cluster-unique. For all objects `A` and `B`, `A.timestamp ≠ B.timestamp`.
+  - Object timestamps are strictly increasing For all objects `A` and `B`, if
+    `A.timestamp < B.timestamp`, then `A` was committed earlier than `B`.
+
```

Comment:
Maybe add something here about consistency guarantees between sessions. Eg if my client crashes and I restart on the same machine, am I guaranteed to see writes from the previous session. If not, how do I deal with that from an application point of view.